mlflow/utils/download_cloud_file_chunk.py

```diff
@@ -0,0 +1,192 @@
+    :param backoff_factor: a time factor for exponential backoff. e.g. value 5 means the HTTP
+      request will be retried with interval 5, 10, 20... seconds. A value of 0 turns off the
+      exponential backoff.
+    :param retry_codes: a list of HTTP response error codes that qualifies for retry.
+    :param timeout: wait for timeout seconds for response from remote server for connect and
+      read request. Default to None owing to long duration operation in read / write.
+    :param kwargs: Additional keyword arguments to pass to `requests.Session.request()`
+
+    :return requests.Response object.
+    """
+    if method.lower() not in ("put", "get", "patch", "delete"):
+        raise ValueError("Illegal http method: " + method)
+    return _get_http_response_with_retries(
+        method,
+        url,
+        max_retries,
+        backoff_factor,
+        retry_codes,
+        timeout=timeout,
+        **kwargs,
+    )
+
+
+def _get_http_response_with_retries(
+    method, url, max_retries, backoff_factor, retry_codes, **kwargs
+):
+    """
+    Performs an HTTP request using Python's `requests` module with an automatic retry policy.
+
+    :param method: a string indicating the method to use, e.g. "GET", "POST", "PUT".
+    :param url: the target URL address for the HTTP request.
+    :param max_retries: Maximum total number of retries.
+    :param backoff_factor: a time factor for exponential backoff. e.g. value 5 means the HTTP
+      request will be retried with interval 5, 10, 20... seconds. A value of 0 turns off the
+      exponential backoff.
+    :param retry_codes: a list of HTTP response error codes that qualifies for retry.
+    :param kwargs: Additional keyword arguments to pass to `requests.Session.request()`
+
+    :return: requests.Response object.
+    """
+    session = _get_request_session(max_retries, backoff_factor, retry_codes)
+    return session.request(method, url, **kwargs)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--range-start", required=True, type=int)
+    parser.add_argument("--range-end", required=True, type=int)
+    parser.add_argument("--headers", required=True, type=str)
+    parser.add_argument("--download-path", required=True, type=str)
+    parser.add_argument("--http-uri", required=True, type=str)
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    try:
+        download_chunk(
+            range_start=args.range_start,
+            range_end=args.range_end,
+            headers=json.loads(args.headers),
+            download_path=args.download_path,
+            http_uri=args.http_uri,
+        )
+    except requests.HTTPError as e:
+        print(  # pylint: disable=print-function
+            json.dumps(
+                {
+                    "error_status_code": e.response.status_code,
+                    "error_text": str(e),
+                }
+            ),
+            file=sys.stdout,
+        )
```

Comment:
This feels like it could be brittle. What if other dependencies print stdout? There must be a better way.